official/utils/testing/golden_test.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""This module tests generic behavior of golden tests.
+
+  This test is not intended to test every layer of interest, and models should
```

Comment:
Hm, you like to indent, I notice. The alignment seems off to me-- am I just wrong on that? I feel like code blocks should be newline separated paragraphs, not indented paragraphs. But that's just me nitting.

Reply:
"followed by the rest of the docstring starting at the same cursor position as the first quote of the first line." You are not wrong.